miner/miner.go

```diff
@@ -192,3 +204,138 @@ func (miner *Miner) SetEtherbase(addr common.Address) {
 func (self *Miner) SubscribePendingLogs(ch chan<- []*types.Log) event.Subscription {
 	return self.worker.pendingLogsFeed.Subscribe(ch)
 }
+
+// commitmentCacheSaved will check to see if the last commitment cache entry is saved.
+// If not, it will search for it and save it.
+func (miner *Miner) commitmentCacheSaved(istEngine consensus.Istanbul) bool {
+	// Subscribe to new block notifications
+	newBlockCh := make(chan core.ChainEvent)
+	bc := miner.eth.BlockChain()
+	newBlockSub := bc.SubscribeChainEvent(newBlockCh)
+	defer newBlockSub.Unsubscribe()
+
+	// getCurrentBlockAndState
+	currentBlock := bc.CurrentBlock()
+	currentHeader := currentBlock.Header()
+	currentState, err := bc.StateAt(currentBlock.Root())
+	if err != nil {
+		log.Error("Error in retrieving state", "block hash", currentHeader.Hash(), "error", err)
+		return false
+	}
+
+	// Check to see if we already have the commitment cache
+	lastCommitment, err := random.GetLastCommitment(miner.coinbase, currentHeader, currentState)
+	if err != nil {
+		log.Error("Error in retrieving last commitment", "error", err)
+		return false
+	}
+
+	// If there was no previous commitment from this validator, then return true.
+	if (lastCommitment == common.Hash{}) {
+		return true
+	}
+
+	if (rawdb.ReadRandomCommitmentCache(miner.db, lastCommitment) != common.Hash{}) {
+		return true
+	}
+
+	// goroutine communication channels and waitgroup
```

Comment:
i think here i would have a new function, something like "regenerateCache"... and a good log message stating taht we are doing such... (and that it might take a while)